modules/percolator/src/main/java/org/elasticsearch/percolator/PercolatorFieldMapper.java

```diff
@@ -346,6 +347,11 @@ static Query toQuery(QueryShardContext context, boolean mapUnmappedFieldsAsStrin
         // as an analyzed string.
         context.setAllowUnmappedFields(false);
         context.setMapUnmappedFieldAsString(mapUnmappedFieldsAsString);
+        // From 5.0 and onwards the percolator rewrites the query at index time,
+        // this is not the case for percolator queries in indices created before 5.0
+        if (context.getIndexSettings().getIndexVersionCreated().before(Version.V_5_0_0_alpha1)) {
+            queryBuilder = QueryBuilder.rewriteQuery(queryBuilder, context);
```

Comment:
It turns out to be a bit more complicated. In case of range query builders this can result in no matches, there the rewrite checks the relation and in case of disjoint returns a match none query builder. The problem is that is uses the index reader from the shard and not from the in-memory index.

Somehow switching to use the in-memory index reader should solve that problem, but that is a much bigger change. So for now I'll keep use the version check.